crates/matrix-sdk-ui/src/timeline/pinned_events_loader.rs

```diff
@@ -64,31 +65,28 @@ impl PinnedEventsLoader {
 
         if !event_ids_to_request.is_empty() {
             let provider = Arc::new(self.room.clone());
-            let mut handles = Vec::new();
 
-            let config = Some(
+            let request_config = Some(
                 RequestConfig::default()
                     .retry_limit(3)
                     .max_concurrent_requests(NonZeroUsize::new(self.max_concurrent_requests)),
             );
 
-            for id in event_ids_to_request {
-                handles.push(tokio::spawn({
-                    let provider = Arc::clone(&provider);
-                    async move {
-                        provider
-                            .event_with_config(&id, config)
-                            .await
-                            .map_err(|_| PinnedEventsLoaderError::EventNotFound(id.to_owned()))
+            let new_events = join_all(event_ids_to_request.into_iter().map(|event_id| {
```

Comment:
I think `join_all` will fail if any of the futures fails, right? In this case we want to fail only if no event could be retrieved, either from the HS or the cache.

Reply:
`join_all` takes a future which output is not a result, so it can't fail. If you look at the `async` block, it doesn log errors. Using `try_join_all` would have caused the issue you describe here.